pkg/kubelet/kubelet.go

```diff
@@ -2848,6 +2857,109 @@ func (kl *Kubelet) fastStatusUpdateOnce() {
 	}, 100*time.Millisecond, stopCh)
 }
 
+// handleMaxContainerCheckpoints goes through the list of checkpoints of a
+// container in a pod and deletes old checkpoints if there are more than
+// 'kl.maxContainerCheckpoints'.
+func (kl *Kubelet) handleMaxContainerCheckpoints(
+	podUID types.UID,
+	podFullName,
+	containerName string,
+	options *runtimeapi.CheckpointContainerRequest,
+) error {
+
+	if kl.maxContainerCheckpoints < 1 {
+		// This cannot really happen as the kubelet config parser will
+		// error out early if this is < 1. This additional error
+		// handling is mainly here to make sure tests are working.
+		return errors.New("maxContainerCheckpoints needs to be >= 1")
+	}
+
+	globPattern := filepath.Join(
+		kl.getCheckpointsDir(),
+		fmt.Sprintf(
+			"checkpoint-%s-%s-*.tar",
```

Comment:
@adrianreber Is it possible for the value of `podFullName` or `containerName`  to contain `-`?

For example, lets assume that we have two pods, each with one container. The name of the first Pod is `foo` and it has a container with name `bar`. The name of the second Pod is `foo-bar` and it has a container with name `test`.

Would this pattern be able to distinguish between the checkpoints of both pods?

Reply:
I extended the glob pattern to match better on the RFC3339 time stamp in the strong which should make sure we do not match on wrong pod/container names.